src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -174,7 +174,17 @@ private function createRelationsForItem($item, $formattedRelations)
                     $relation->dissociate()->save();
                 }
             } elseif ($relation instanceof HasOne || $relation instanceof MorphOne) {
-                $modelInstance = $relation->updateOrCreate([], $relationDetails['values']);
+                $relation_values = $relationDetails['values'][$relationMethod] ?? $relationDetails['values'] ?? null;
+
+                if ($relation_values === null) {
+                    $relation->delete();
```

Comment:
I couldn't delete a related `hasOne` entry. 

I have:

```php
        CRUD::field('passport')->fields([
            ['name' => 'number', 'type' => 'number'],
            ['name' => 'first_name', 'type' => 'text', 'wrapperAttributes' => ['class' => 'form-group col-md-4']],
            ['name' => 'middle_name', 'type' => 'text', 'wrapperAttributes' => ['class' => 'form-group col-md-4']],
            ['name' => 'last_name', 'type' => 'text', 'wrapperAttributes' => ['class' => 'form-group col-md-4']],
            ['name' => 'birth_date', 'type' => 'date'],
            ['name' => 'notes', 'type' => 'textarea'],
            ['name' => 'issuance_date', 'type' => 'date', 'wrapperAttributes' => ['class' => 'form-group col-md-6']],
            ['name' => 'expiry_date', 'type' => 'date', 'wrapperAttributes' => ['class' => 'form-group col-md-6']],
        ]);
```

And when I remove the passport from the pet (X on the repeatable item), upon editing the same entry I see the passport has NOT been deleted:

![2022-01-11 11 51 28](https://user-images.githubusercontent.com/1032474/148920370-5f8050b8-977c-4672-9a33-04aaa2bbea68.gif)

Creating worked fine (wonderfully actually), editing worked fine, but when I delete the entry from repeatable, I expect it to be deleted... that's the whole point we've created this `relationship.entry` field, right?

Can you reproduce this @pxpm ? Was this intentional? Am I missing something?

Reply:
Totally, I missed it, I made it work first then had a look at tests and they were failing for the `dot.notation` so I fixed them and unintentionally broke this.

I've already submitted the PR to fix this, but don't merge it yet I will be adding tests in a few moments for this scenario too!

Reply:
Confirmed, it works now 🎉